services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/team/TeamManagerImpl.java

```diff
@@ -707,6 +707,28 @@ public String getIconURL(UserInfo userInfo, String teamId) throws NotFoundExcept
 		return fileHandleManager.getRedirectURLForFileHandle(urlRequest);
 	}
 
+	@Override
+	public String getIconPreviewURL(UserInfo userInfo, String teamId) throws NotFoundException {
+		Team team = teamDAO.get(teamId);
```

Comment:
Let us add some input validation, e.g. using the ValidateArgument.required... methods.